### PR TITLE
Logfmt: add support for new structure and new features (flags, arguments)

### PR DESCRIPTION
**Pull request by matyax**
This PR does a small refactor to Logfmt related grammar to sync it with https://github.com/grafana/loki/blob/main/pkg/logql/syntax/expr.y, and to support new Logfmt features:
- Flags: --strict --keep-empty
- Arguments: label1, label2, label4="label3"

With these changes, new types of nodes have been introduced:

- LogfmtParser
- LogfmtExpressionParser (formerly included as a LabelParser)
- LogfmtParserFlags
- LabelExtractionExpressionList (former JsonExpressionList)
- LabelExtractionExpression (former JsonExpression)

New trees:

simple tree

full tree

Part of https://github.com/grafana/grafana/issues/71121

### CODE REVIEW

File: src/logql.grammar [hunk 1]
```diff
@@ -397,6 +414,10 @@ ConvOp {
     "`" ![`]* "`"
   }
 
+  ParserFlag {
+    "--strict" | "--keep-empty"
```
Comment (ivanahuckova) [inline src/logql.grammar:418]: Have you considered saving these as keywords? Just curious, I am not sure if that would be better.

File: test/expression.txt [hunk 2]
```diff
@@ -436,7 +436,7 @@ LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier, Eq, String))), Pipeline
 
 ==>
 
-LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier, Eq, String))), PipelineExpr(PipelineExpr(PipelineExpr(PipelineStage(Pipe, LabelParser(Logfmt))), PipelineStage(Pipe, LabelFilter(Matcher(Identifier, Eq, String)))), PipelineStage(Pipe, DecolorizeExpr(Decolorize))))))
+LogQL(Expr(LogExpr(Selector(Matchers(Matcher(Identifier, Eq, String))), PipelineExpr(PipelineExpr(PipelineExpr(PipelineStage(Pipe, LogfmtParser(Logfmt))), PipelineStage(Pipe, LabelFilter(Matcher(Identifier, Eq, String)))), PipelineStage(Pipe, DecolorizeExpr(Decolorize))))))
 
 # Range aggregation expression - "(" Selector ")" Range OffsetExpr UnwrapExpr
```
Comment (ivanahuckova) [inline test/expression.txt:441]: Would you add some tests for logql with parser flags?

File: src/logql.grammar (hunk 1 above)
Comment (ivanahuckova) [reply src/logql.grammar:418]: I tried to implement it by adding `--strict` as tokens, but I can't make it work. I was just curios, but this works as expected also for query such as `{label="value"} |= "--strict" | logfmt --strict` so LGTM.

Comment (svennergr) [reply src/logql.grammar:418]: Question would be if we want to further distinguish the flags by lezer? E.g. any use cases where we need to know that it's a query with a `StrictParserFlag`?

File: src/formatter/logs.js [hunk 3]
```diff
@@ -30,6 +30,7 @@ import {
   Selector,
   PipelineExpr,
 } from '../parser.js';
+import { LogfmtParser } from '../parser.terms.js';
```
Comment (ivanahuckova) [inline src/formatter/logs.js:33]: Should this be imported from `'../parser.js';` as everything else?

File: src/logql.grammar (hunk 1 above)
Comment (ivanahuckova) [reply src/logql.grammar:418]: We could still do it by checking the value inside `ParserFlag` node.

**Review by ivanahuckova — APPROVED** [review]

**Pull request merged**